ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -786,8 +788,9 @@ garbageCollectKeys :: String -> Int -> CheckParents -> [(Key, Int)] -> Action [K
 garbageCollectKeys label maxAge checkParents agedKeys = do
     start <- liftIO offsetTime
     extras <- getShakeExtras
-    (n::Int, garbage) <- liftIO $ modifyVar (state extras) $ \vmap ->
-        evaluate $ foldl' removeDirtyKey (vmap, (0,[])) agedKeys
+    let values = state extras
+    (n::Int, garbage) <- liftIO $ atomically $
```

Comment:
given your previous changes involved pushing `atomically` inside lops, should this one also be inside the loop?

Reply:
Yeah, but it also needs to be done atomically with `removeDirtyKey`.